### PR TITLE
BL-6096 A11y Checks auto refresh

### PR DESCRIPTION
**Pull request by hatton**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496)

### CODE REVIEW

**Comment by andrew-polk** [comment]
Reviewed 14 of 14 files at r1.
Review status: all files reviewed, 3 unresolved discussions (waiting on @hatton)

---

*[src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx, line 10 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzCe699ZImIvi8WHkP:-LEzCe699ZImIvi8WHkQ:bkg9bmh) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/24e83c6040b4c86ada831826d109b47f67ad09ca/src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx#L10)):*
> ```XML
>     // increments it whenever it wants us to re-query
>     // the server.
>     refreshCount: number;
> ```

Clearly I am missing something. I don't see how/where this is used. I see that C# can tell the parent to refresh and the parent can tell the child by setting this. But what does the child do with this? And is it an int rather than a boolean so we don't have to try to reset it back to false when we are done?

---

*[src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx, line 38 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzB78i30undpiVvzmL:-LEzB78i30undpiVvzmM:b2urjc8) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/24e83c6040b4c86ada831826d109b47f67ad09ca/src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx#L38)):*
> ```
> refreshGeneration
> ```

refreshCount

---

*[src/BloomBrowserUI/react_components/apiBackedCheckbox.tsx, line 14 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzB9rO7fhO8fEnUSVu:-LEzB9rO7fhO8fEnUSVv:b1d9ast) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/24e83c6040b4c86ada831826d109b47f67ad09ca/src/BloomBrowserUI/react_components/apiBackedCheckbox.tsx#L14)):*
> ```XML
>     // increments it whenever it wants us to re-query
>     // the server.
>     refreshGeneration?: number;
> ```

Probably want this to be refreshCount like the other components? (and change comment below)

---


*Comments from [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496)*

**Comment by andrew-polk** [comment]
Reviewed 4 of 4 files at r2.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---

*[src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx, line 10 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzCe699ZImIvi8WHkP:-LEzGnTIAiCwRAcAKiFi:b-l3lbux) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/24e83c6040b4c86ada831826d109b47f67ad09ca/src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx#L10)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Yes... I was hoping the comment on line 37 would be enough, but maybe not if you're not used to react?

>  // We're using componentWillReceiveProps() instead of componentWillMount so that we
>    // can re-query if/when our refreshCount number changes.
</blockquote></details>

Ok, looking at the docs so I can understand better.
But I see this method is now deprecated. We should use https://reactjs.org/docs/react-component.html#componentdidupdate instead. (assuming our version is new enough.)

---


*Comments from [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496)*

**Comment by gmartin7** [comment]
:lgtm:

---

Reviewed 11 of 14 files at r1, 4 of 4 files at r2.
Review status: all files reviewed, 1 unresolved discussion (waiting on @hatton)

---

*[src/BloomBrowserUI/react_components/apiBackedCheckbox.tsx, line 27 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzRfSB4bKux5eAuFwL:-LEzRfSC8msc7Kyxy1M0:b-rg00p0) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/686c00fa30d08308d58f062a41368f024c4de8a7/src/BloomBrowserUI/react_components/apiBackedCheckbox.tsx#L27)):*
> ```XML
>     // We're using can re-query() instead of componentWillMount so that we can re-query
>     // if/when our refreshCount number changes.
>     public componentWillReceiveProps() {
> ```

Something weird about this comment. Do you mean "We're using componentWillReceiveProps..."?

---


*Comments from [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-:-LEzT4-U4QLIMT-r0yBW:bnfp4nl)*

**Comment by hatton** [comment]
Review status: all files reviewed, 2 unresolved discussions (waiting on @hatton and @gmartin7)

---

*[src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx, line 10 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzCe699ZImIvi8WHkP:-LEzfukdCag6ajegOxhX:b-dm7m61) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/24e83c6040b4c86ada831826d109b47f67ad09ca/src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx#L10)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Ok, looking at the docs so I can understand better.
But I see this method is now deprecated. We should use https://reactjs.org/docs/react-component.html#componentdidupdate instead. (assuming our version is new enough.)
</blockquote></details>

Yeah, looks like in some future version they rename it to unsafe. I don't agree with their analysis in this case... that's fairly common in react-land, that if you aren't doing it "the facebook way", they sorta freak out, documentation-wise.

---

*[src/BloomBrowserUI/react_components/apiBackedCheckbox.tsx, line 27 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzRfSB4bKux5eAuFwL:-LEzfKUw4boc8OYx-zFn:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/686c00fa30d08308d58f062a41368f024c4de8a7/src/BloomBrowserUI/react_components/apiBackedCheckbox.tsx#L27)):*
<details><summary><i>Previously, gmartin7 (Gordon Martin) wrote…</i></summary><blockquote>

Something weird about this comment. Do you mean "We're using componentWillReceiveProps..."?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496)*

**Comment by andrew-polk** [comment]
Review status: all files reviewed, 1 unresolved discussion (waiting on @gmartin7)

---

*[src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx, line 10 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzCe699ZImIvi8WHkP:-LEzhUeQ1p8yCkQGmTCj:b-rq9g10) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/24e83c6040b4c86ada831826d109b47f67ad09ca/src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx#L10)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Yeah, looks like in some future version they rename it to unsafe. I don't agree with their analysis in this case... that's fairly common in react-land, that if you aren't doing it "the facebook way", they sorta freak out, documentation-wise.
</blockquote></details>

Ok, trusting your analysis. But this is beyond a disagreement with the documentation. This will actually be a breaking change when we upgrade versions of react. Probably worth a comment explaining why you think we should use this -- with unsafe_ -- even after the upgrade (if that is what you're saying) or what it should be changed to at that time.
In other words, it would be good not to lose the analysis you've already done.

---

*[src/BloomBrowserUI/react_components/apiBackedCheckbox.tsx, line 27 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzRfSB4bKux5eAuFwL:-LEzhSJI78mZA8K5V7vc:b25mvej) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/686c00fa30d08308d58f062a41368f024c4de8a7/src/BloomBrowserUI/react_components/apiBackedCheckbox.tsx#L27)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

This change didn't seem to get pushed.

---


*Comments from [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496)*

**Comment by hatton** [comment]
Review status: all files reviewed, 1 unresolved discussion (waiting on @gmartin7)

---

*[src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx, line 10 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzCe699ZImIvi8WHkP:-LEziI1m1JUU4rV0qnwl:bai2af8) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/24e83c6040b4c86ada831826d109b47f67ad09ca/src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx#L10)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Ok, trusting your analysis. But this is beyond a disagreement with the documentation. This will actually be a breaking change when we upgrade versions of react. Probably worth a comment explaining why you think we should use this -- with unsafe_ -- even after the upgrade (if that is what you're saying) or what it should be changed to at that time.
In other words, it would be good not to lose the analysis you've already done.
</blockquote></details>

Oh fine, andrew, I'll change it.

---


*Comments from [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496)*

**Comment by hatton** [comment]
Review status: all files reviewed, 1 unresolved discussion (waiting on @gmartin7)

---

*[src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx, line 10 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-LEzCe699ZImIvi8WHkP:-LEzlwO7CngcGSdj-mC1:b-bol5vj) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/24e83c6040b4c86ada831826d109b47f67ad09ca/src/BloomBrowserUI/publish/accessibilityCheck/checkItem.tsx#L10)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Oh fine, andrew, I'll change it.
</blockquote></details>

Sigh, componentDidUpdate() lead to infinite recursion, of course. Hmmmm

---


*Comments from [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496)*

**Comment by andrew-polk** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r3.
Dismissed @gmartin7 from a discussion.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/2496#-:-LF-RTky6uy7LxLuunVx:bnfp4nl)*
